habitat_sim/simulator.py

```diff
@@ -284,6 +284,9 @@ def apply_torque(self, torque, object_id, scene_id=0):
     def get_world_time(self, scene_id=0):
         return self._sim.get_world_time()
 
+    def recompute_navmesh(self, navmesh_settings):
+        self.pathfinder = self._sim.recompute_navmesh(navmesh_settings)
```

Comment:
IMO, it makes more sense to have the pathfinder be recomputed in place. i.e. `self._sim.recompute_navmesh` should take a reference to the current pathfinder and return nothing.  This'll keep weird bugs from happening where the new pathfinder doesn't get picked up by something else that has a reference to the old one.

Reply:
`self.pathfinder.load_nav_mesh` loads the navmesh from a file which was generated with some fixed settings, so new settings cannot be used.

Original intent was to allow the navmesh to be stored for the case that you want multi-agent support or some other use of multiple navmeshes with different settings.

Reply:
Due to nice division of responsibilities, something like `self.pathfinder.recompute_navmesh(navmesh_settings)` isn't possible without giving the pathfinder way more information than it needs access to do its job 99% of the time.

Reply:
I still think `self._sim.recompute_navmesh(self.pathfinder, navmesh_settings)` is cleaner and less likely to run into weird bugs where the changes to the pathfinder don't get propagated (i.e. something still has a reference to the old pathfinder).

Reply:
Sure, so we alter the function to take an existing pathfinder and reseat it, rather than returning a new one. Would still suite my purpose of allowing multiple pathfinders to exist. I'll make this change. 👍

Reply:
Sounds good!  You can of course make an overloaded version for all these things that creates a new pathfinder, but since this method is modifying `self.pathfinder` it seems cleaner to do it in place :)